### PR TITLE
Disallow double bracket initialization

### PR DESCRIPTION
**Pull request by willmostly**
Fixes https://github.com/trinodb/trino-gateway/issues/155.

### CODE REVIEW

**Review by mosabua — APPROVED** [review]
Looks good to me at first glance.

**Comment by mosabua** [comment]
@ebyhr ?

File: gateway-ha/src/main/java/io/trino/gateway/ha/clustermonitor/TrinoQueueLengthChecker.java [hunk 1]
```diff
@@ -48,18 +49,8 @@ public void observe(List<ClusterStats> stats)
...
-                            }
-                        });
-                clusterRunningMap.put(stat.getRoutingGroup(), new HashMap<String, Integer>()
-                        {
-                            {
-                                put(stat.getClusterId(), stat.getRunningQueryCount());
-                            }
-                        });
+                clusterQueueMap.put(stat.getRoutingGroup(), ImmutableMap.of(stat.getClusterId(), stat.getQueuedQueryCount()));
+                clusterRunningMap.put(stat.getRoutingGroup(), ImmutableMap.of(stat.getClusterId(), stat.getRunningQueryCount()));
```
Comment (ebyhr) [inline gateway-ha/src/main/java/io/trino/gateway/ha/clustermonitor/TrinoQueueLengthChecker.java:53]: Is this change really safe? I can't find the existing test coverage. `else` block calls `ImmutableMap.put` which isn't supported.

Is it possible to use Table class in Guava?

File: gateway-ha/src/test/java/io/trino/gateway/ha/security/TestLbAuthenticator.java [hunk 2]
```diff
@@ -107,13 +107,10 @@ public void testAuthenticatorMissingClaim()
...
-        {
-            {
-                this.put("user1", new UserConfiguration("priv1, priv2", "pass1"));
-                this.put("user2", new UserConfiguration("priv2, priv2", "pass2"));
-            }
-        };
+        Map presetUsers = ImmutableMap.of("user1",
+                new UserConfiguration("priv1, priv2", "pass1"),
+                "user2",
+                new UserConfiguration("priv2, priv2", "pass2"));
```
Comment (ebyhr) [inline gateway-ha/src/test/java/io/trino/gateway/ha/security/TestLbAuthenticator.java:113]: nit: Avoid raw usage of parameterized map and wrap per entry:
```java
        Map<String, UserConfiguration> presetUsers = ImmutableMap.of(
                "user1", new UserConfiguration("priv1, priv2", "pass1"),
                "user2", new UserConfiguration("priv2, priv2", "pass2"));
```
Same for another change.

File: pom.xml [hunk 3]
```diff
@@ -122,11 +122,8 @@
                     <failOnWarning>false</failOnWarning>
                     <compilerArgs>
                         <arg>-XDcompilePolicy=simple</arg>
-                        <arg>
-                            -Xplugin:ErrorProne
-                            <!-- TODO: Enable DoubleBraceInitialization flag -->
-                            -Xep:DoubleBraceInitialization:OFF
-                        </arg>
+                        <arg>-Xplugin:ErrorProne
+                             -Xep:DoubleBraceInitialization:ERROR</arg>
```
Comment (ebyhr) [inline pom.xml:126]: The default severity of `DoubleBraceInitialization` is ERROR. https://errorprone.info/bugpattern/DoubleBraceInitialization

```suggestion
                        <arg>-Xplugin:ErrorProne</arg>
```

File: gateway-ha/src/main/java/io/trino/gateway/ha/clustermonitor/TrinoQueueLengthChecker.java (hunk 1 above)
Comment (willmostly) [reply gateway-ha/src/main/java/io/trino/gateway/ha/clustermonitor/TrinoQueueLengthChecker.java:53]: Good catch - will update

Comment (willmostly) [reply gateway-ha/src/main/java/io/trino/gateway/ha/clustermonitor/TrinoQueueLengthChecker.java:53]: Clarification - are you suggesting replacing the `Map<String, Map<String, Int>` with a `Table<String, String, Int>`? 

It doesn't look like there is a test case that covers the `else`. Maybe `TestClusterStatsMonitor` could be extended to test this

Comment (ebyhr) [reply gateway-ha/src/main/java/io/trino/gateway/ha/clustermonitor/TrinoQueueLengthChecker.java:53]: Right - https://github.com/google/guava/wiki/CollectionUtilitiesExplained#tables

**Review by ebyhr — APPROVED** [review]

**Pull request merged**